tests/test_app/test_sqlite_data_store.py

```diff
@@ -260,26 +260,30 @@ def test_read_log(sql_dstore, log_data):
     assert got == log_data
 
 
-def test_write(sql_dstore, ro_dir_dstore):
-    """correctly write content"""
+@pytest.mark.parametrize("binary", (False, True))
+def test_write_text_binary(sql_dstore, ro_dir_dstore, binary):
+    """correctly write content whether text or binary data"""
+    db = SqliteDataStore(":memory:", mode=OVERWRITE)
     expect = Path(ro_dir_dstore.source / "brca1.fasta").read_text()
-    identifier = "result_table/brca1.fasta"
-    sql_dstore.write(unique_id=identifier, data=expect)
-    got = sql_dstore.read(identifier)
+    if binary:
+        expect = dumps(expect)
+    identifier = "brca1.fasta"
+    m = db.write(unique_id=identifier, data=expect)
+    got = m.read()
     assert got == expect
 
 
 def test_write_if_member_exists(sql_dstore, ro_dir_dstore):
     """correctly write content"""
     expect = Path(ro_dir_dstore.source / "brca1.fasta").read_text()
-    identifier = "result_table/brca1.fasta"
-    sql_dstore.write(unique_id=identifier, data=expect)
-    got = sql_dstore.read(identifier)
+    identifier = "brca1.fasta"
+    m = sql_dstore.write(unique_id=identifier, data=expect)
+    got = m.read()
     assert got == expect
     sql_dstore._mode = OVERWRITE
-    identifier = "result_table/brca1.fasta"
-    sql_dstore.write(unique_id=identifier, data=expect)
-    got = sql_dstore.read(identifier)
+    identifier = "brca1.fasta"
+    m = sql_dstore.write(unique_id=identifier, data=expect)
+    got = m.read()
```

Comment:
Yesterday I had problems here, because `write` function had problem. `write` function returns a new `DataMember` after updating and adds it to the datastore, so we will have two similar members, but it should return None when it updates a DataMember.
I have done two fixes for this, in my code:
1 -  I fixed `write` function to return None after updating . So it won't add new member to datastore.
2 - I have added
`assert len_dstore == len(full_dstore)`
here to check updating not adding new member.

Reply:
I have these changes in my code. You don't need to do anything.

Reply:
I don't think it should ever return `None`. If a member already exists in the collection, then it should return that instance

Reply:
@GavinHuttley 
Because after calling `_write`, it adds new member to the list. if we return the existing member, it will add it as a new member.